addons/arsenal/functions/fnc_attributeAddCompatible.sqf

```diff
@@ -23,51 +23,36 @@ private _category = lbCurSel (_controlsGroup controlsGroupCtrl IDC_ATTRIBUTE_CAT
 // Exit if selected category is not attachments or magazines
 if !(_category in [4, 5, 6, 7, 8]) exitWith {};
 
-private _configItems = +(uiNamespace getVariable [QGVAR(configItems), []]);
+private _configItems = uiNamespace getVariable [QGVAR(configItems), []];
 private _attributeValue = uiNamespace getVariable [QGVAR(attributeValue), [[], 0]];
 _attributeValue params ["_attributeItems"];
 
 // Get list of all weapons in attribute items
-(_configItems select 0) params ["_primaryWeapons", "_secondaryWeapons", "_handgunWeapons"];
-private _attributeWeapons = _attributeItems select {_x in _primaryWeapons || {_x in _secondaryWeapons} || {_x in _handgunWeapons}};
+private _attributeWeapons = _attributeItems arrayIntersect (flatten (_configItems select IDX_VIRT_WEAPONS));
 
 // Add compatible attachments or magazines to attribute
-private _cfgWeapons = configFile >> "CfgWeapons";
 private _itemsToAdd = [];
 
 if (_category == 8) then {
-    private _magazineGroups = uiNamespace getVariable QGVAR(magazineGroups);
-    private _cfgMagazines = configFile >> "CfgMagazines";
-
+    // Get all compatible magazines for weapons
     {
-        private _weaponConfig = _cfgWeapons >> _x;
-
-        {
-            private _muzzleConfig = if (_x == "this") then {_weaponConfig} else {_weaponConfig >> _x};
-
-            // Only add existent magazines and ensure correct classname case
-            private _magazines = getArray (_muzzleConfig >> "magazines") select {isClass (_cfgMagazines >> _x)};
-            _magazines = _magazines apply {configName (_cfgMagazines >> _x)};
-            _itemsToAdd append _magazines;
-
-            {
-                _itemsToAdd append (_magazineGroups get (toLower _x));
-            } forEach getArray (_muzzleConfig >> "magazineWell");
-        } forEach getArray (_weaponConfig >> "muzzles");
+        _itemsToAdd append (compatibleMagazines _x);
```

Comment:
I understand if the ACE team wants `CBA_fnc_compatibleMagazines` to be used instead (because of ongoing discussions on the CBA GitHub).

Reply:
If we do essentially the same thing, its better to use CBA to not have duplicate code
https://community.bistudio.com/wiki/compatibleMagazines is also a thing now, don't know what CBA uses

Reply:
I looked at your CBA PR.
Assuming your PR is merged,
Con:
This would do the same thing but without cache (is this performance critical?) and filter (which isn't needed) so it would be faster/simpler not to use CBA.
Pro:
If any bug is found in compatibleMagazines command, CBA could temp fix it in their function. But you don't get access to that if you don't use it here.

Reply:
And one important thing on your CBA PR it was said that the engine command can return duplicates? (I should fix that btw)
That can be bad we don't want to display duplicate items in arsenal view.

Reply:
Decision has been taken to move forward with engine commands instead of CBA. If bugs are found with engine commands, ACE will revert back to the CBA functions.